### PR TITLE
Some SpimDataOpener Refactoring

### PR DESCRIPTION
**Pull request by tischi**

### CODE REVIEW

**Comment by tischi** [comment]
@KateMoreva 

Please wait with the review, I am still working on it. 

I will ping you when I am done!

**Comment by KateMoreva** [comment]
> @KateMoreva
> 
> Please wait with the review, I am still working on it.
> 
> I will ping you when I am done!

@tischi 

I just merged the develop brunch as I was not aware that you were also working on this code and made some changes. Sorry for interrupting.

**Comment by tischi** [comment]
Ah, OK, I though you wanted to merge my branch!

What were you working on?

Maybe wait until we merged my branch to avoid conflicts...

**Comment by KateMoreva** [comment]
> Ah, OK, I though you wanted to merge my branch!
> 
> What were you working on?
> 
> Maybe wait until we merged my branch to avoid conflicts...

I fixed the issue with bdv.hdf5 in the `develop` branch (as it is an important functionality).

**Comment by tischi** [comment]
OK, thanks!

Maybe please don't work on it until we merge this PR as we did exactly the same thing and I had to fix the conflicts 😄

**Comment by tischi** [comment]
@KateMoreva @K-Meech I think this is ready for review. Unfortunately I cannot find back the issue with the list of things that we wanted to do (one of them was to move many things from imagej-utils to mobie-io). Anyway, I think I did all of it.

Could you please have a look? Ideally we would merge today such that I could give this to Tobias Pietzsch, who wanted to have a look...

It will be a bit of work to make mobie-viewer-fiji compatible with this version, but we can do this later.
I copied some classes that I deleted from mobie-io to the tischi branch of mobie-viewer-fiji such that they are not lost: https://github.com/mobie/mobie-viewer-fiji/pull/566

File: pom.xml
Comment (KateMoreva) [inline pom.xml:8]: Could you also update the  <artifactId>mobie-io</artifactId> version, please? I think that it would fail to build with the old one.

**Review by KateMoreva — APPROVED** [review]

**Comment by KateMoreva** [comment]
> @KateMoreva @K-Meech I think this is ready for review. Unfortunately I cannot find back the issue with the list of things that we wanted to do (one of them was to move many things from imagej-utils to mobie-io). Anyway, I think I did all of it.
> 
> Could you please have a look? Ideally we would merge today such that I could give this to Tobias Pietzsch, who wanted to have a look...
> 
> It will be a bit of work to make mobie-viewer-fiji compatible with this version, but we can do this later. I copied some classes that I deleted from mobie-io to the tischi branch of mobie-viewer-fiji such that they are not lost: [mobie/mobie-viewer-fiji#566](https://github.com/mobie/mobie-viewer-fiji/pull/566)

I checked what is needed to be done on the mobie-viewer-fiji side. I think that I might fix it today too.

File: pom.xml
Comment (tischi) [reply pom.xml:8]: Done.

**Pull request merged**
